src/lando/headless_api/tests/test_automation_api.py

```diff
@@ -0,0 +1,427 @@
+        response.status_code == 202
+    ), "Successful submission should result in `202 Accepted` status code."
+
+    response_json = response.json()
+
+    job_id = response_json["job_id"]
+    assert isinstance(job_id, int), "Job ID should be returned as an `int`."
+
+    assert response_json["status_url"] == f"https://lando.test/api/job/{job_id}"
+    assert response_json["message"] == "Job is in the SUBMITTED state."
+    assert is_isoformat_timestamp(
+        response_json["created_at"]
+    ), "Response should include an ISO formatted creation timestamp."
+
+    job = AutomationJob.objects.get(id=job_id)
+
+    for index, action in enumerate(job.actions.all()):
+        assert action.data["content"] == str(
+            index
+        ), "Actions should be retrieved in order of submission."
+
+
+@pytest.mark.django_db
+def test_get_job_status_not_found(client, headless_user):
+    user, token = headless_user
+    response = client.get(
+        "/api/job/12345",
+        headers={
+            "User-Agent": "testuser@example.org",
+            "Authorization": f"Bearer {token}",
+        },
+    )
+    assert (
+        response.status_code == 404
+    ), "API should respond with a 404 for non-existent job ID."
+
+
+@pytest.mark.parametrize(
+    "status,message",
+    (
+        (LandingJobStatus.SUBMITTED, "Job is in the SUBMITTED state."),
+        (LandingJobStatus.IN_PROGRESS, "Job is in the IN_PROGRESS state."),
+        (LandingJobStatus.DEFERRED, "Job is in the DEFERRED state."),
+        (LandingJobStatus.FAILED, "Job is in the FAILED state."),
+        (LandingJobStatus.LANDED, "Job is in the LANDED state."),
+        (LandingJobStatus.CANCELLED, "Job is in the CANCELLED state."),
+    ),
+)
+@pytest.mark.django_db
+def test_get_job_status(status, message, client, headless_user):
+    user, token = headless_user
+
+    # Create a job and actions
+    job = AutomationJob.objects.create(status=status)
+    AutomationAction.objects.create(
+        job_id=job, action_type="add-commit", data={"content": "test"}, order=0
+    )
+
+    # Fetch job status.
+    response = client.get(
+        f"/api/job/{job.id}",
+        headers={
+            "User-Agent": "testuser@example.org",
+            "Authorization": f"Bearer {token}",
+        },
+    )
+
+    assert (
+        response.status_code == 200
+    ), "Response code should be 200 when status is retrieved successfully."
+
+    response_data = response.json()
+
+    assert response_data["job_id"] == job.id
+    assert (
+        response_data["message"] == message
+    ), "Response message should align with current job status."
+    # TODO test a few more things? formatting?
+
+
+PATCH_NORMAL_1 = r"""
+# HG changeset patch
+# User Test User <test@example.com>
+# Date 0 0
+#      Thu Jan 01 00:00:00 1970 +0000
+# Diff Start Line 7
+add another file.
+diff --git a/test.txt b/test.txt
+--- a/test.txt
++++ b/test.txt
+@@ -1,1 +1,2 @@
+ TEST
++adding another line
+""".strip()
```

Comment:
We can use the `normal_patch` fixture https://github.com/mozilla-conduit/lando/blob/ff0dbd3c06af3830975e3f3e70f6949b9c964938/src/lando/api/tests/conftest.py#L131-L143to get this one out, rather than duplicating it.